tests/test_data_processing.py

```diff
@@ -26,14 +38,512 @@
+        # Check whether reading a DataFrame missing required columns errors out
+        load_scalars(path_file_missing_required)
+
+
+def test_load_timeseries():
+    """
+    This test checks whether
+    1.  the DataFrame read by load_timeseries function from data which is a
+            a. time series with multiIndex
+            b. sequence
+            c. stacked time series / sequence
+        contains all default columns and
+    2.  load_timeseries errors out if data is missing a required column
+
+    """
+    # 1. Test
+    cols_list = [
+        "id_ts",
+        "region",
+        "var_name",
+        "timeindex_start",
+        "timeindex_stop",
+        "timeindex_resolution",
+        "series",
+        "var_unit",
+        "source",
+        "comment",
+    ]
+
+    # a. time series with multiIndex
+    path_file_timeseries = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_timeseries.csv",
+    )
+
+    # b. sequence
+    path_file_sequence = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_sequence.csv",
+    )
+
+    # c. stacked time series / sequence
+    path_file_stacked = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_stacked.csv",
+    )
+
+    paths = [
+        path_file_timeseries,
+        path_file_sequence,
+        path_file_stacked,
+    ]
+
+    # Run 1. Test for formats a., b. and c.
+    for path_file in paths:
+        df = load_timeseries(path_file)
+        df_cols = list(df.columns)
+
+        assert df_cols == cols_list
+
+    # 2. Test
+    path_file_stacked_missing_required = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_stacked_missing_required.csv",
+    )
+
+    with pytest.raises(KeyError):
+        # Check whether reading a stacked DataFrame missing required columns errors out
+        load_timeseries(path_file_stacked_missing_required)
+
+
+def test_save_df():
+    """
+    This test checks for scalars and time series whether
+    1. the path of the scalars stored in a csv file exists
+    2. the DataFrame remain unchanged after saving. For this purpose, the data is read in again
+    after saving and compared with the scalars originally read
+
+    """
+    # Scalars
+    path_file_scalars = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_scalars.csv",
+    )
+
+    path_file_saved = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_scalars_saved.csv",
+    )
+
+    # Read scalars
+    df = load_scalars(path_file_scalars)
+
+    # Save read scalars
+    save_df(df, path_file_saved)
+
+    # Load the saved scalars
+    df_saved = load_scalars(path_file_saved)
+
+    # 1. Test
+    assert os.path.exists(path_file_saved) == 1
```

Comment:
This seems obsolete as the test would fail in the previous line if the file would not exist.

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/57/commits/10429062c7180571172a6d83d4628f3da593b5ba.